### PR TITLE
Status service cloudformation

### PR DESCRIPTION
**Pull request by bhasin85**
Sample command to test the cloudformation. Please replace stack and domain name.

`aws cloudformation deploy --template-file wps-cloudformation-template.yaml --stack-name job-status-test6 --parameter-overrides requestHandlerFilename=request-handler-0.01-lambda-package.zip requestHandlerCodeURL=https://s3-ap-southeast-2.amazonaws.com/wps-lambda/request-handler-0.01-lambda-package.zip jobStatusFilename=job-status-service-0.01-lambda-package.zip jobStatusCodeURL=https://s3-ap-southeast-2.amazonaws.com/wps-lambda/job-status-service-0.01-lambda-package.zip dockerImage=615645230945.dkr.ecr.ap-southeast-2.amazonaws.com/javaduck:sandbox geoserver=http://geoserver-123.aodn.org.au/geoserver/imos/ows acmCertificate=arn:aws:acm:us-east-1:615645230945:certificate/06e50848-29a7-47c3-8a3f-c3212b608a8d hostedZoneName=dev.aodn.org.au requestHandlerDomainName=request-handler6 jobStatusDomainName=job-status6 --capabilities CAPABILITY_IAM`

### CODE REVIEW

**Comment by jonescc** [comment]
Would be good if this worked with deploy.sh as well:

An error occurred (ValidationError) when calling the CreateChangeSet operation: Parameters: [jobStatusFilename, requestHandlerFilename] must have values

**Comment by jonescc** [comment]
Actually, looks those two params need to be added to the example dev.properties and s3LambdaCodeFilename removed.   I'll update my own properties file.

**Comment by bhasin85** [comment]
@jonescc  properties file


```
requestHandlerFilename=request-handler-0.01-lambda-package.zip
requestHandlerCodeURL=https://s3-ap-southeast-2.amazonaws.com/wps-lambda/request-handler-0.01-lambda-package.zip
jobStatusFilename=job-status-service-0.01-lambda-package.zip
jobStatusCodeURL=https://s3-ap-southeast-2.amazonaws.com/wps-lambda/job-status-service-0.01-lambda-package.zip 
dockerImage=615645230945.dkr.ecr.ap-southeast-2.amazonaws.com/javaduck:sandbox
geoserver=http://geoserver-123.aodn.org.au/geoserver/imos/ows
acmCertificate=arn:aws:acm:us-east-1:615645230945:certificate/06e50848-29a7-47c3-8a3f-c3212b608a8d
hostedZoneName=dev.aodn.org.au
requestHandlerDomainName=request-handler6
jobStatusDomainName=job-status6
```

**Comment by jonescc** [comment]
As discussed, do we need two api gateway's - perhaps another resource such as wps/jobStatus would do?

**Comment by jonescc** [comment]
Looks good to me - tested OK.  @akashisama, @lwgordonimos any comments?  I'm not a cloudformation expert so perhaps someone could run their eyes over that.

File: wps-cloudformation-template.yaml [hunk 1]
```diff
@@ -538,11 +561,12 @@ Resources:
...
           DESCRIBE_PROCESS_S3_KEY_PREFIX: !Ref s3configKey
-          GEOSERVER_WPS_ENDPOINT_URL: !If [CreateCustomDomainName, !Join ['', ['https://', !Ref domainName, '.', !Ref hostedZoneName,'/wps']], !Join ['', ['https://', !Ref WPSApi, '.execute-api.',!Ref "AWS::Region",'.amazonaws.com','/LATEST/wps']]]
+          GEOSERVER_WPS_ENDPOINT_URL: !If [CreateWpsApiGatewayDomainName, !Join ['', ['https://', !Ref wpsDomainName, '.', !Ref hostedZoneName,'/wps']], !Join ['', ['https://', !Ref WPSRestApi, '.execute-api.',!Ref "AWS::Region",'.amazonaws.com','/', !Ref wpsApiStage,'/', !Ref requestHandlerApiPath]]]
           AWS_REGION_SES: !Ref awsRegionSes
           JOB_S3_KEY: !Ref s3jobKey
+          STATUS_SERVICE_ENDPOINT: !If [CreateWpsApiGatewayDomainName, !Join ['', ['https://', !Ref wpsDomainName, '.', !Ref hostedZoneName,'/', !Join ['/', [!Ref requestHandlerApiPath, !Ref jobStatusApiPath]]]], !Join ['', ['https://', !Ref WPSRestApi, '.execute-api.',!Ref "AWS::Region",'.amazonaws.com','/', !Ref wpsApiStage,'/', !Join ['/', [!Ref requestHandlerApiPath, !Ref jobStatusApiPath]]]]]
```
Comment (akashisama) [inline wps-cloudformation-template.yaml]: Putting joins within joins seems unnecessary convoluted here

**Pull request merged**